corgidrp/l4_to_tda.py

```diff
@@ -139,6 +139,81 @@ def convert_to_flux(input_dataset, fluxcal_factor):
+        fluxcal_factor (corgidrp.data.FluxcalFactor): flux calibration file
+        photo (String): do either aperture photometry ("aperture") or 2DGaussian fit of the point source ("2dgauss")
+        phot_kwargs (dict): parameters of the photometry method, for details see fluxcal.aper_phot and fluxcal.phot_by_gauss_2dfit
+
+    Returns:
+        corgidrp.data.Dataset: a version of the input dataset with the data in flux units
+    """
+   # you should make a copy the dataset to start
+    flux_dataset = input_dataset.copy()
+    if "COL_COR" in flux_dataset[0].ext_hdr:
+        color_cor_fac = flux_dataset[0].ext_hdr['COL_COR']
+    else: 
+        warnings.warn("There is no COL_COR keyword in the header, color correction was not done, it is set to 1")
+        color_cor_fac = 1
+    factor = fluxcal_factor.fluxcal_fac/color_cor_fac
+    factor_error = fluxcal_factor.fluxcal_err/color_cor_fac
+    if photo == "aperture":
+        if phot_kwargs == None:
+            #set default values
+            phot_kwargs = {
+              'encircled_radius': 5,
+              'frac_enc_energy': 1.0,
+              'method': 'subpixel',
+              'subpixels': 5,
+              'background_sub': False,
+              'r_in': 5,
+              'r_out': 10,
+              'centering_method': 'xy',
+              'centroid_roi_radius': 5
+            }
+        phot_values = [fluxcal.aper_phot(image, **phot_kwargs) for image in flux_dataset]
+    elif photo == "2dgauss":
+        if phot_kwargs == None:
+            #set default values
+            phot_kwargs = {
+              'fwhm': 3,
+              'fit_shape': None,
+              'background_sub': False,
+              'r_in': 5,
+              'r_out': 10,
+              'centering_method': 'xy',
+              'centroid_roi_radius': 5
+              }
+        phot_values = [fluxcal.phot_by_gauss2d_fit(image, **phot_kwargs) for image in flux_dataset]
+    else:
+        raise ValueError(photo + " is not a valid photo parameter, choose aperture or 2dgauss")
+    
+    if phot_kwargs.get('background_sub', False):
+        ap_sum, ap_sum_err, back = np.mean(np.array(phot_values),0)
+    else:
+        ap_sum, ap_sum_err = np.mean(np.array(phot_values),0)
+        back = 0
+    
+    flux = ap_sum * factor
+    flux_err = np.sqrt(ap_sum_err**2 * factor**2 + factor_error**2 *ap_sum**2)
+    #Also determine the apparent Vega magnitude
+    filter_file = fluxcal.get_filter_name(flux_dataset[0])
+    vega_mag = fluxcal.calculate_vega_mag(flux, filter_file)
+    
+    history_msg = "star {0} flux calculated as {1} erg/(s * cm^2 * AA) corresponding to {2} vega magnitude".format(flux_dataset[0].pri_hdr["TARGET"],flux, vega_mag)
+
+    # update the output dataset with this converted data and update the history
+    flux_dataset.update_after_processing_step(history_msg, header_entries = {"FLUXFAC": fluxcal_factor.fluxcal_fac, "LOCBACK": back, "FLUX": flux, "FLUXERR": flux_err, "APP_MAG": vega_mag})
```

Comment:
would it be easy to calculate the error on the app mag? if not, please make an issue for us to do it later.